src/modules/expenses/expense.test.ts

```diff
@@ -0,0 +1,229 @@
+      fullName: "Test User",
+    });
+
+    testUser2 = await createTestUser({
+      email: "testUser2@sample.com",
+      password: "12345678",
+      role: AuthRoles.USER,
+      fullName: "Test User",
+    });
+
+    adminUser = await createTestUser({
+      email: "adminUser@sample.com",
+      password: "12345678",
+      role: AuthRoles.ADMIN,
+      fullName: "Test Admin",
+    });
+
+    testCategory = await createTestCategory({ name: "Test category" });
+    testCategory2 = await createTestCategory({ name: "Test category2", userId: testUser2.id });
+  });
+
+  describe("POST /expenses", () => {
+    it("should create an expense as user", async () => {
+      const response = await expenseClient.expenses.$post(
+        {
+          json: {
+            amount: 100,
+            currency: "USD",
+            splitType: "even",
+            description: "expense description",
+            categoryId: testCategory.id,
+          },
+        },
+        {
+          headers: {
+            session: testUser.session,
+          },
+        },
+      );
+
+      if (response.status === HTTPStatusCodes.CREATED) {
+        const json = await response.json();
+
+        expect(json.success).toBe(true);
+        expect(json.message).toBe("Expense created successfully");
+        expect(json.data).toHaveProperty("amount", 100);
+        expect(json.data).toHaveProperty("currency", "USD");
+        expect(json.data).toHaveProperty("splitType", "even");
+        expect(json.data).toHaveProperty("categoryId", testCategory.id);
+        expect(json.data).toHaveProperty("payerId", testUser.id);
+      }
+    });
+
+    it("should create an expense as admin", async () => {
+      const response = await expenseClient.expenses.$post(
+        {
+          json: {
+            amount: 100,
+            currency: "USD",
+            splitType: "even",
+            description: "expense description",
+            payerId: testUser.id,
+            categoryId: testCategory.id,
+          },
+        },
+        {
+          headers: {
+            session: adminUser.session,
+          },
+        },
+      );
+
+      if (response.status === HTTPStatusCodes.CREATED) {
+        const json = await response.json();
+
+        expect(json.success).toBe(true);
+        expect(json.message).toBe("Expense created successfully");
+        expect(json.data).toHaveProperty("amount", 100);
+        expect(json.data).toHaveProperty("currency", "USD");
+        expect(json.data).toHaveProperty("splitType", "even");
+        expect(json.data).toHaveProperty("creatorId", adminUser.id);
+        expect(json.data).toHaveProperty("categoryId", testCategory.id);
+        expect(json.data).toHaveProperty("payerId", testUser.id);
+      }
+    });
+
+    it("should return 400 when admin creates expense without providing payer id", async () => {
+      const response = await expenseClient.expenses.$post(
+        {
+          json: {
+            amount: 100,
```

Comment:
These values are reused can be extracted to a single object at the top and referenced everywhere

Reply:
Done. I moved common fields to an object `expenseTestCommonFields`